doc/man7/ssl.pod

```diff
@@ -1,883 +0,0 @@
-in this object.
-
-When a network connection has been created, it can be assigned to an
-B<SSL> object. After the B<SSL> object has been created using
-L<SSL_new(3)>, L<SSL_set_fd(3)> or
-L<SSL_set_bio(3)> can be used to associate the network
-connection with the object.
-
-When the TLS/SSL handshake is performed using
-L<SSL_accept(3)> or L<SSL_connect(3)>
-respectively.
-L<SSL_read_ex(3)>, L<SSL_read(3)>, L<SSL_write_ex(3)> and L<SSL_write(3)> are
-used to read and write data on the TLS/SSL connection.
-L<SSL_shutdown(3)> can be used to shut down the
-TLS/SSL connection.
-
-=head1 DATA STRUCTURES
-
-Currently the OpenSSL B<ssl> library functions deals with the following data
-structures:
-
-=over 4
-
-=item B<SSL_METHOD> (SSL Method)
-
-This is a dispatch structure describing the internal B<ssl> library
-methods/functions which implement the various protocol versions (SSLv3
-TLSv1, ...). It's needed to create an B<SSL_CTX>.
-
-=item B<SSL_CIPHER> (SSL Cipher)
-
-This structure holds the algorithm information for a particular cipher which
-are a core part of the SSL/TLS protocol. The available ciphers are configured
-on a B<SSL_CTX> basis and the actual ones used are then part of the
-B<SSL_SESSION>.
-
-=item B<SSL_CTX> (SSL Context)
-
-This is the global context structure which is created by a server or client
-once per program life-time and which holds mainly default values for the
-B<SSL> structures which are later created for the connections.
-
-=item B<SSL_SESSION> (SSL Session)
-
-This is a structure containing the current TLS/SSL session details for a
-connection: B<SSL_CIPHER>s, client and server certificates, keys, etc.
-
-=item B<SSL> (SSL Connection)
-
-This is the main SSL/TLS structure which is created by a server or client per
-established connection. This actually is the core structure in the SSL API.
-At run-time the application usually deals with this structure which has
-links to mostly all other structures.
-
-=back
-
-
-=head1 HEADER FILES
-
-Currently the OpenSSL B<ssl> library provides the following C header files
-containing the prototypes for the data structures and functions:
-
-=over 4
-
-=item B<ssl.h>
-
-This is the common header file for the SSL/TLS API.  Include it into your
-program to make the API of the B<ssl> library available. It internally
-includes both more private SSL headers and headers from the B<crypto> library.
-Whenever you need hard-core details on the internals of the SSL API, look
-inside this header file.
-
-=item B<ssl2.h>
-
-Unused. Present for backwards compatibility only.
-
-=item B<ssl3.h>
-
-This is the sub header file dealing with the SSLv3 protocol only.
-I<Usually you don't have to include it explicitly because
-it's already included by ssl.h>.
-
-=item B<tls1.h>
-
-This is the sub header file dealing with the TLSv1 protocol only.
-I<Usually you don't have to include it explicitly because
-it's already included by ssl.h>.
-
-=back
```

Comment:
Aside from the opening sentence of the "DESCRIPTION" section which needs updating, I think everything above this point in the file still seems to hold some value - so should be kept (and probably extended or rewritten to be made better at some point (not this PR)). Everything below this point is fairly useless and should be removed.

Reply:
I think that keeping just the top part is a reasonable compromise.  Since the function declarations are removed, do you have any thoughts on removing the SEE ALSO links that I did?

Reply:
I think pointing at the overview page from other SSL related pages seems like a reasonable thing to do, so I would keep the SEE ALSO links.

Reply:
I want to push back on that a bit.  There are roughly 135 SSL manpages but only 100 of them link to ssl(7).  I think I should either add the missing links or keep them as currently removed.
```
; ls SSL*|wc
    135     135    3511
; grep 'ssl(7)' SSL* | wc
    105     186    5495
```

(Yes I know that's counting files, not those with multiple API's in them, which is why I said "roughly")

If we just look for '(7)' among all manpages the ratio is still about one-third:
```
; g grep '(7)' | wc    
    305     694   15384
; ls | wc
    496     496   10820
```

So add ssl(7) to all SSL pages, keep it with the current partial, or remove ssl(7) altogether?

Reply:
I don't particularly feel the need to add `ssl(7)` SEE ALSO links to the remaining SSL man pages.  But I certainly wouldn't object if you did it.

Reply:
What the L it was only 30 files, so I added it. :)

(Get the joke?)